CMakeLists.txt

```diff
@@ -1,177 +1,225 @@
-    message("Debugnet turned on")
-    add_definitions(-DDEBUGNET)
-    target_link_libraries(${VITA_ELF_NAME} debugnet)
-    file(STRINGS "debugnetip.txt" DEBUGNETIP)
-    add_definitions(-DDEBUGNETIP="${DEBUGNETIP}")
-  endif()
-endif(CMAKE_BUILD_TYPE MATCHES Debug)
-
-target_link_libraries(${VITA_ELF_NAME}
-  yaml-cpp
-  m
-  vita2d
-  SceDisplay_stub
-  SceGxm_stub
-  SceSysmodule_stub
-  SceCtrl_stub
-  SceTouch_stub
-  ScePgf_stub
-  SceCommonDialog_stub
-  freetype
-  png
-  jpeg
-  z
-  m
-  c
-  SceNet_stub
-  SceNetCtl_stub
-  SceHttp_stub
-  SceSsl_stub
-  ${CURLPP_LDFLAGS}
-  ${CURL_LIBRARIES}
-  ${OPENSSL_LIBRARIES}
-  ftpvita
-  SceAppMgr_stub
-  SceAppUtil_stub
-  ScePromoterUtil_stub
-  SceIme_stub
-  ScePower_stub
-  SceAudio_stub
-  SceAudiodec_stub
-  SceVshBridge_stub
-  pthread
-)
-
-vita_create_self(${SHORT_NAME}.self ${VITA_ELF_NAME} UNSAFE UNCOMPRESSED)
-vita_create_vpk(${SHORT_NAME}.vpk ${VITA_TITLEID} ${SHORT_NAME}.self
-  VERSION ${VITA_VERSION}
-  NAME ${VITA_APP_NAME}
-  FILE sce_sys/icon0.png sce_sys/icon0.png
-  FILE sce_sys/livearea/contents/bg.png sce_sys/livearea/contents/bg.png
-  FILE sce_sys/livearea/contents/startup.png sce_sys/livearea/contents/startup.png
-  FILE sce_sys/livearea/contents/template.xml sce_sys/livearea/contents/template.xml
-
-  FILE assets/icons.zip resources/icons.zip
-  FILE assets/fonts/segoeui.ttf resources/fonts/segoeui.ttf
-)
-
-get_filename_component(psvitaipFilePath psvitaip.txt REALPATH)
-if(EXISTS "${psvitaipFilePath}")
-  file(STRINGS "${psvitaipFilePath}" PSVITAIP)
-
-  add_custom_target(send
-    COMMAND curl -T ${SHORT_NAME}.self ftp://${PSVITAIP}:1337/ux0:/app/${VITA_TITLEID}/eboot.bin
-    DEPENDS ${SHORT_NAME}.self
-  )
-
-  add_custom_target(shellsend
-    COMMAND psp2shell_cli ${PSVITAIP} 3333 load ${VITA_TITLEID} ${SHORT_NAME}.self
-    DEPENDS ${SHORT_NAME}.self
-  )
-
-  add_custom_target(vpksend
-    COMMAND curl -T ${SHORT_NAME}.vpk ftp://${PSVITAIP}:1337/ux0:/
-    DEPENDS ${SHORT_NAME}.vpk
-  )
-else()
-  message("Couldn't find psvitaip.txt. Put the IP-address of you vita there to get new build targets.")
-endif()
+cmake_minimum_required(VERSION 2.8)
+
+if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+  if(DEFINED ENV{VITASDK})
+    set(CMAKE_TOOLCHAIN_FILE "$ENV{VITASDK}/share/vita.toolchain.cmake" CACHE PATH "toolchain file")
+  else()
+    message(FATAL_ERROR "Please define VITASDK to point to your SDK path!")
+  endif()
+endif()
+
+
+set(SHORT_NAME VitaHBBrowser)
+project(${SHORT_NAME})
+include("${VITASDK}/share/vita.cmake" REQUIRED)
+
+set(ENV{PKG_CONFIG_PATH} "$ENV{VITASDK}/arm-vita-eabi/lib/pkgconfig")
+include(FindPkgConfig)
+
+set(VITA_APP_NAME "Vita Homebrew Browser")
+set(VITA_TITLEID "VHBB00001")
+set(UPDATER_TITLEID "VHBB00002")
+set(VITA_VERSION "00.83")
+
+
+set(PACKAGE_TEMP_FOLDER "ux0:/temp/pkg/")
+
+add_subdirectory(src_updater)
+create_updater("${SHORT_NAME}" "${VITA_TITLEID}" ${UPDATER_TITLEID} "${PACKAGE_TEMP_FOLDER}")
+
+add_custom_command(

```

Comment:
See, the main idea (also behind copying the debug stuff) was to make the updater an independent and easy-to-reuse app. So if you don't copy the binaries you don't have control over the symbols' names because `ld` derives the names from the input file's name and there is nothing you can do about it. And I preferred a more generic symbol like `_binary_assets_updater_eboot_bin_start` instead of `_binary_VitaHBBrowserUpdater_param_sfo_start` which is very project specific.

Reply:
Ah I see, in this case I'd suggest you to add them to `vita_create_vpk` and `fopen()` them where you need it.

If it's not possible, then please add a comment in `CMakeLists.txt` to quickly explain why you copy them.

Reply:
This might be a good idea. Only drawback I see is that it slows down the dev process as you have to re-install the `vpk` every time the updater changes (as it isn't part of the `eboot` anymore). Am I wrong? If I'm right, do you have any solution to this or would you prefer the clarity of `fopen` over the ease of development?

Reply:
This is true, but you can still send the files to the right place using FTP (just like you do with the eboot.bin currently), you don't actually need to reinstall the VPK.

Reply:
Ok so I'll just have to update my tools. Sound easy enough.
One more concern that just came to my mind: are other apps able to write inside the resource folder of VHBB? If so, this were a potential attack for arbitrary code execution.

Reply:
So xyz confirmed that all sysapp can write to any folder. This makes embedding the updater app in the binary the safer choice, I'd say. Even though it is still attackable as other sys apps could of course manipulate VHBB's binary. You're opinion @devnoname120?